client/src/components/flashcard/flashcardModal.jsx

```diff
@@ -8,24 +8,33 @@ import {
   ListItemSuffix,
   IconButton,
 } from "@material-tailwind/react";
-import { PlusIcon, TrashIcon } from "@heroicons/react/24/solid";
+import { BoltIcon, PlusIcon, TrashIcon } from "@heroicons/react/24/solid";
 import React, { useEffect, useState } from "react";
-import { useSelector } from "react-redux";
+import { useDispatch, useSelector } from "react-redux";
 import Cards from "./flashcards";
 import AddModuleModal from "./addModuleModal";
 import DeleteModal from "./confirmDeleteModal";
 import { Player } from "@lottiefiles/react-lottie-player";
+import SraIntroModal from "./sraIntroModal";
+import SmartCards from "./smartStudyCarousel";
+import { getScheduledCardsAsync } from "../../store/flashcards/thunks";
 
 
 const FlashCardModal = (props) => {
   const modules = useSelector((state) => state.flashcards.modules);
+  const user = useSelector((state) => state.loginReducer);
   const [id, setId] = useState(props.moduleId);
   const [key, setKey] = useState(props.moduleId);
   const [visible, setVisible] = useState(false);
+  const [sraVisible, setSraVisible] = useState(false);
+  const [showSra, setShowSra] = useState(false);
   const [deleteVisible, setDeleteVisible] = useState(false);
 
+  const dispatch = useDispatch();
+
   useEffect(() => {
     setId(props.moduleId);
+    setKey(props.moduleId);
   }, [props.moduleId]);
```

Comment:
Why do we need both id and key? Would the key is enough?

Reply:
So, If you look closely at the whole file, you'll notice that key and id both have different purposes. key is being incremented in a useEffect to facilitate reset functionality for a module whereas id is not being modified like that, an id is just being used to track which module a user is at from a list of modules.